system/lib/wasmfs/backends/opfs_backend.cpp

```diff
@@ -118,6 +118,32 @@ void _wasmfs_opfs_flush_access(em_proxying_ctx* ctx, int access_id, int* err);
 namespace {
 
 using ProxyWorker = emscripten::ProxyWorker;
+using ProxyingQueue = emscripten::ProxyingQueue;
+
+class Worker {
+public:
+#ifdef __EMSCRIPTEN_PTHREADS__
+  ProxyWorker proxy;
+
+  template<typename T>
+  void operator()(T func) {
+    proxy(func);
+  }
+#else
+  // When used with JSPI on the main thread the various wasmfs_opfs_* functions
+  // can be directly executed since they are all async.
+  template <typename T>
+  void operator()(T func) {
+    if constexpr (std::is_invocable<T&, ProxyingQueue::ProxyingCtx>::value) {
```

Comment:
Very slightly simpler:
```suggestion
    if constexpr (std::is_invocable_v<T&, ProxyingQueue::ProxyingCtx>) {
```